### PR TITLE
DEP: special.factorial: raise error for non-integer scalars and `exact=True`

### PR DESCRIPTION
**Pull request by j-bowhay**
follow up to https://github.com/scipy/scipy/commit/18bb04b95df98d5264bd3e522b8adde916f17e05, scheduled for this release

### CODE REVIEW

File: scipy/special/_basic.py [hunk 1]
```diff
@@ -3020,8 +3020,8 @@ def factorial(n, exact=False):
             return math.factorial(n)
         elif exact:
             msg = ("Non-integer values of `n` together with `exact=True` are "
```
Comment (j-bowhay) [inline scipy/special/_basic.py:3022]:
```suggestion
            msg = ("Non-integer values of `n` together with `exact=True` is "
```

Comment (h-vetinari) [reply scipy/special/_basic.py:3022]: I prefer this as-is (with "are", which refers to plural "values").

**Review by h-vetinari — APPROVED** [review]
Thanks for doing this clean-up!

File: scipy/special/_basic.py (hunk 1 above)
Comment (j-bowhay) [reply scipy/special/_basic.py:3022]: Yes I agree, clearly too tired when I wrote that!

**Review by lucascolley — APPROVED** [review]
thanks Jake!

**Comment by lucascolley** [comment]
@j-bowhay @h-vetinari it looks like the same thing for `factorialk` is scheduled for this release

**Pull request merged**